src/debug_commads.c

```diff
@@ -665,6 +665,50 @@ DEBUG_COMMAND(GCForceBGInvoke) {
   return REDISMODULE_OK;
 }
 
+DEBUG_COMMAND(GCStopFutureRuns) {
+  if (argc < 1) {
+    return RedisModule_WrongArity(ctx);
+  }
+  StrongRef ref = IndexSpec_LoadUnsafe(ctx, RedisModule_StringPtrLen(argv[0], NULL), 0);
+  IndexSpec *sp = StrongRef_Get(ref);
+  if (!sp) {
+    return RedisModule_ReplyWithError(ctx, "Unknown index name");
+  }
+  GCTask *task;
+  if (RedisModule_StopTimer(RSDummyContext, sp->gc->timerID, (void **)&task) == REDISMODULE_OK) {
+    rm_free(task);
```

Comment:
This task might have a blocked client? If so lets unblock it with an error?

Reply:
The periodic task never has a blocked client. I can add a verification here but only the blocking debug command use the field and adds its own task. It also has a timeout for the blocking so it won’t get stuck
Let me know if you’d still prefer that I’ll add a verification

Reply:
No, seems ok.